api/src/main/java/org/pmiops/workbench/api/ConceptsController.java

```diff
@@ -65,16 +69,37 @@
             .vocabularyId(concept.getVocabularyId())
             .conceptSynonyms(concept.getSynonyms().stream().map(ConceptSynonym::getConceptSynonymName).collect(Collectors.toList()));
 
+  static final Function<DbDomain, DomainInfo> TO_CLIENT_DOMAIN_INFO =
+      (dbDomain) -> new DomainInfo()
+          .domain(Domain.fromValue(dbDomain.getDomainId().toUpperCase()))
+          .name(dbDomain.getDomainDisplay())
```

Comment:
I could imagine doing this. If so I'd want to switch DbDomain to use our Domain enum, having the separate counts for standard concepts and all concepts... even if we only populate the standard counts in data browser. But given that the UI code is different, the endpoints are different, and the API implementation is different, it wasn't clear to me what the benefit of sharing the API model object would be.

Reply:
Discussed offline: I'm going to put DomainInfo in common-api, and we can either in this PR or in a follow-up change data browser to use it. The objects are similar enough (and likely to remain similar enough in future) that sharing the model definition doesn't seem to hurt anything. We might be able to share a little bit of API code, too (e.g. converting database entities to DomainInfo.)

Reply:
@biopete One wrinkle with this plan... I was hoping to use the Domain enum for domains in this object (as I did with DomainInfo.) This is the enum we use when creating concept sets; it's important that the domains we show in the concept search UI in workbench match up with those domains. The current list of domain_id values in db_domain is:

+----------------+
| domain_id      |
+----------------+
| Condition      |
| CPT4           |
| Drug           |
| Fam Med His    |
| ICD10          |
| ICD10CM        |
| ICD9CM         |
| Lifestyle      |
| Measurement    |
| Overall Health |
| Per Med His    |
| Procedure      |
| RxNorm         |
| SNOMED         |
+----------------+

This includes a bunch of values that aren't in the enum. So I'm not sure what to do with them.

We could not use an enum here, having a string instead as you do in the DbInfo object, but this makes it more likely we'll return some stuff we can't use with concept sets.

Reply:
I should note: we do have values in Domain already that we won't use for concept sets -- and we could add enum values for the rest of these domain IDs, too. But I think most of these values (e.g. CPT4, ICD9, the individual questionnaire modules) don't translate to what we're considering a "domain" in the workbench code... they are vocabularies or groupings of concepts within a domain.

Reply:
@danqrodney Is this the Domain enum you are talking about ? https://github.com/all-of-us/workbench/blob/2035091eb560ae179e2cf0994ac5e01163170759/api/src/main/java/org/pmiops/workbench/api/ConceptsController.java#L32

So this table can be changed. It is a WIP imo. Feel free to use this table and redesign it or whatever as long as it can still serve our basic purpose. It will get imported into workbench from the generate-cdr/csv/db_domain.csv .. We don't mind changing a little databrowser code.

It  was used to specify vocabulary filters as well like what you need on the concept set pages.  But we don't use them now. We can take all the vocabs out from that table and it won't hurt us. We used to have a domain_parent column that specified what vocabularies were in what domain so we could show the filters for that domain.  

Maybe if we add a vocubulary_id column , along with the db_type column you can use to solve your needs. The demain_route column we don't use.  Omop , does not put any relationship data in concept_relationship for vocabulary and domain thus we just made it our selves in this table and could control what filters we needed to display and the display text and such.

You can always add more rows to get what you want with a specific new db_type ... 

Hope that helps.

Reply:
OK. So I'd like to remove all the vocabulary stuff from this table (sounds like it's not in use.)

Putting PPI modules in db_domain also feels a bit wrong... they aren't really "domains" like Measurement, Condition, Procedure, and Drug, and the domain_id value doesn't really mean anything (like it does for the others.) I can imagine us wanting extra metadata for PPI modules (e.g. links to the surveys on the web, date that a PPI module was published, order that modules are presented / shown in the UI, etc.) They are going to get updated more often (as new PPI modules come along). We are surfacing some similar fields on them in the current data browser UX, but I kinda feel like they deserve their own ppi_module table and response object in the API; we can fetch from both tables and return two lists in the same endpoint if that's useful. How does that sound to you?

Reply:
I'll note that in general, the current PPI UX where modules and questions are displayed in arbitrary order and not grouped by topics, is not going to make sense to researchers. If we can't generate the right UX just from OMOP relationships, we're going to need to have a separate configuration for PPI stuff, updated on a per CDR release basis. This could be another database table; but it could just be a configuration JSON thing, similar to:

https://github.com/all-of-us/workbench/blob/master/api/config/cdm/cdm_5_2.json

and could potentially get generated from our codebook.

Assuming we do need some configuration of this stuff, I'm tempted to say that JSON is going to be easier to manage than database tables, as we already have command line tools for updating JSON configs in different environments, and we should for now at least list PPI modules in configuration rather than a DB table. And having a hierarchical database structure for representing modules and questions within them that can be updated is probably more complicated to implement.